lib/loader/index.js

```diff
@@ -65,6 +65,32 @@ function instantiate(module, imports) {
     return parts.join("") + String.fromCharCode.apply(String, U16.subarray(dataOffset, dataOffset + dataRemain));
   }
 
+  /** Gets an array from the module's memory by its pointer and its type. */
+  function getArray(ptr, type, asTippedArray) {
+    checkMem();
+    const ptr0 = ptr >>> 2;
+    const dataStart = (U32[ptr0] >>> 2) + 2; // 0 -> size; 1 -> static 0
+    const len = U32[ptr0 + 1];
+    const transform = asTippedArray ? (x) => x : (x) => Array.from(x);
+    const recast = asTippedArray ? (x) => type.constructor.from(x) : (x) => x;
+    if (type === U16 || type == U8) {
+      return recast(U32.subarray(dataStart, dataStart + len));
+    }
+    if (type === I16 || type == I8) {
+      return recast(I32.subarray(dataStart, dataStart + len));
```

Comment:
I think this isn't correct, as it slices 32-bit values from memory even if the value is 8- or 16-bit in memory. You could actually use the `U16` etc. views to slice from, and just return the subarray.

Reply:
I am not certain how to get it here. When I look at `const arr: Array<u16>= [1, 2, 3]` in the U16 view, I get 
```[
...,
12,
0,
0,
0,
1, // data start
0,
2,
0,
3,
0,
...
]
So using subarray will get me at best `[1, 0, 2, 0, 3, 0]`.

Reply:
An element of an U16 view is 16-bit respectively 2 bytes. Hence, you'd slice it like `U16.subarray(dataStart >>> 1, (dataStart >>> 1) + arrayLength)`.

Reply:
I pushed a change in that direction, but it actually break the tests. I'll be back at it after a good night sleep.